modules/auxiliary/gather/progress_moveit_sftp_fileread_cve_2024_5806.rb

```diff
@@ -0,0 +1,202 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+require 'net/ssh/transport/session'
+require 'net/sftp'
+require 'openssl'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Progress MOVEit SFTP Authentication Bypass for Arbitrary File Read',
+        'Description' => %q{
+          This module exploits CVE-2024-5806, an authentication bypass vulnerability in the MOVEit Transfer SFTP service. The
+          following version are affected:
+
+          * MOVEit Transfer 2023.0.x (Fixed in 2023.0.11)
+          * MOVEit Transfer 2023.1.x (Fixed in 2023.1.6)
+          * MOVEit Transfer 2024.0.x (Fixed in 2024.0.2)
+
+          The module can establish an authenticated SFTP session for a MOVEit Transfer user. The module allows for both listing
+          the contents of a directory, and the reading of an arbitrary file.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'sfewer-r7' # MSF Module & Rapid7 Analysis
+        ],
+        'References' => [
+          ['CVE', '2024-5806'],
+          ['URL', 'https://attackerkb.com/topics/44EZLG2xgL/cve-2024-5806/rapid7-analysis'] # AttackerKB Rapid7 Analysis.
+        ],
+        'DisclosureDate' => '2024-06-25',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RHOST,
+        Opt::RPORT(22),
+        OptBool.new('STORE_LOOT', [false, 'Store the target file as loot', true]),
+        OptString.new('TARGETUSER', [true, 'A valid username to authenticate as.', nil]),
+        OptString.new('TARGETFILE', [true, 'The full path of a target file or directory to read.', '/'])
```

Comment:
I get this error:
```bash
>set Proxies tcp:127.0.0.1:3333
[!] Unknown datastore option: Proxies.
Proxies => tcp:127.0.0.1:3333
```
Are we missing the Proxies option?
```suggestion
        Opt::RHOST,
        Opt::RPORT(22),
        OptBool.new('STORE_LOOT', [false, 'Store the target file as loot', true]),
        OptString.new('TARGETUSER', [true, 'A valid username to authenticate as.', nil]),
        OptString.new('TARGETFILE', [true, 'The full path of a target file or directory to read.', '/']),
        Opt::Proxies
```

Reply:
Good catch, yes it appears we must explicitly register an `Opt::Proxies` option. Your above patch looks good, I can commit that